This file contains diffs of checklink revision 1d91dde (March 3, 2017),
against the modified version of checklink that appears in this directory.

diff -u -b -r /home/mernst/bin/src/checklink-w3c/bin/checklink-20170303 /home/mernst/bin/src/checklink-w3c/bin/checklink
--- /home/mernst/bin/src/checklink-w3c/bin/checklink-20170303	2018-09-21 06:04:50.493256542 -0700
+++ /home/mernst/bin/src/checklink-w3c/bin/checklink	2018-09-21 06:07:06.675596989 -0700
@@ -23,6 +23,8 @@
 
 use strict;
 use 5.008;
+use English;
+$WARNING = 1;
 
 # Get rid of potentially unsafe and unneeded environment variables.
 delete(@ENV{qw(IFS CDPATH ENV BASH_ENV)});
@@ -45,7 +47,7 @@
 use Net::HTTP::Methods 5.833 qw();    # >= 5.833 for 4kB cookies (#6678)
 
 # if 0, ignore robots exclusion (useful for testing)
-use constant USE_ROBOT_UA => 1;
+use constant USE_ROBOT_UA => 0;
 
 if (USE_ROBOT_UA) {
     @W3C::UserAgent::ISA = qw(LWP::RobotUA);
@@ -221,6 +223,7 @@
 use CSS::DOM::Style qw();
 use CSS::DOM::Util qw();
 use Encode qw();
+use File::Basename;
 use HTML::Entities qw();
 use HTML::Parser 3.40 qw();    # >= 3.40 for utf8_mode()
 use HTTP::Headers::Util qw();
@@ -1246,6 +1249,16 @@
             }
         }
 
+        if ($abs_link_uri eq "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd") {
+          $abs_link_uri = URI->new("file://" . dirname(__FILE__) . "/../etc/xhtml1-transitional.dtd");
+        }
+        if ($abs_link_uri eq "http://www.w3.org/TR/html4/frameset.dtd") {
+          $abs_link_uri = URI->new("file://" . dirname(__FILE__) . "/../etc/frameset.dtd");
+        }
+        if ($abs_link_uri eq "http://www.w3.org/TR/REC-html40/loose.dtd") {
+          $abs_link_uri = URI->new("file://" . dirname(__FILE__) . "/../etc/loose.dtd");
+        }
+
         my $canon_uri = URI->new($abs_link_uri->canonical());
         my $fragment  = $canon_uri->fragment(undef);
         if (!defined($Opts{Exclude}) || $canon_uri !~ $Opts{Exclude}) {

Diff finished.  Fri Sep 21 06:07:19 2018
